micro_dl/inference/image_inference.py

```diff
@@ -669,94 +676,100 @@ def predict_2d(self, chan_slice_meta):
-                    model=self.model,
-                    input_image=cur_input,
-                )
-            for i, chan_idx in enumerate(self.target_channels):
-                meta_row = chan_meta.loc[chan_meta['channel_idx'] == chan_idx, :].squeeze()
-                if self.model_task == 'regression':
-                    pred_image[:, i, ...] = self.unzscore(pred_image[:, i, ...],
-                                                   cur_target[:, i, ...],
-                                                   meta_row)
-                # save prediction
-                self.save_pred_image(
-                    im_input=cur_input,
-                    im_target=cur_target[:, i:i+1, ...],
-                    im_pred=pred_image[:, i:i+1, ...],
-                    meta_row=meta_row,
-                    pred_chan_name=self.pred_chan_names[i]
-                )
+                    pred_image = inference.predict_large_image(
+                        model=self.model,
+                        input_image=cur_input,
+                    )
 
-            # get mask
-            if self.mask_metrics:
-                cur_mask = self.get_mask(meta_row[0])
-                mask_stack.append(cur_mask)
-            # add to vol
-            pred_stack.append(pred_image)
-            target_stack.append(cur_target.astype(np.float32))
+                for i, chan_idx in enumerate(self.target_channels):
+                    meta_row = chan_meta.loc[chan_meta['channel_idx'] == chan_idx, :].squeeze()
+                    if self.model_task == 'regression':
+                        if self.input_depth > 1:
+                            pred_image[:, i, 0, ...] = self.unzscore(pred_image[:, i, 0, ...],
+                                                           cur_target[:, i, 0, ...],
+                                                           meta_row)
+                        else:
+                            pred_image[:, i, ...] = self.unzscore(pred_image[:, i, ...],
+                                                           cur_target[:, i, ...],
+                                                           meta_row)
+
+                # get mask
+                if self.mask_metrics:
+                    cur_mask = self.get_mask(chan_meta[0])
+                    mask_stack.append(cur_mask)
+                # add to vol
+                input_stack.append(cur_input)
+                pred_stack.append(pred_image)
+                target_stack.append(cur_target.astype(np.float32))
+                pbar.update(1)
+
+        input_stack = np.concatenate(input_stack, axis=0)
         pred_stack = np.concatenate(pred_stack, axis=0) #zcyx
         target_stack = np.concatenate(target_stack, axis=0)
         # Stack images and transpose (metrics assumes cyxz format)
         if self.image_format == 'zyx':
             if self.input_depth > 1:
+                input_stack = input_stack[:, :, 0, :, :]
```

Comment:
The center slice for input should be selected here

Reply:
Done!